internal/pkg/service/buffer/task/orchestrator/orchestrator.go

```diff
@@ -143,19 +147,23 @@ func Start[T any](ctx context.Context, wg *sync.WaitGroup, d dependencies, confi
 	return w.start(ctx, wg)
 }
 
-func (w orchestrator[R]) start(ctx context.Context, wg *sync.WaitGroup) <-chan error {
-	work := func(distCtx context.Context, assigner *distribution.Assigner) <-chan error {
-		return w.config.Source.WatchPrefix.
-			GetAllAndWatch(distCtx, w.client, w.config.Source.WatchEtcdOps...).
-			SetupConsumer(w.logger).
+func (o orchestrator[R]) start(ctx context.Context, wg *sync.WaitGroup) <-chan error {
+	work := func(ctx context.Context, assigner *distribution.Assigner) <-chan error {
+		ctx, span := o.tracer.Start(ctx, SpanNamePrefix+"."+o.config.Name)
+		return o.config.Source.WatchPrefix.
+			GetAllAndWatch(ctx, o.client, o.config.Source.WatchEtcdOps...).
+			SetupConsumer(o.logger).
+			WithOnClose(func(err error) {
+				telemetry.EndSpan(span, &err)
+			}).
```

Comment:
Pridany `orchestrator` span + `EndSpan` cez `WithOnClose`.

Tento `GetAllAndWatch` sa pravidelne resetuje, vid `distribution.WithResetInterval(o.config.Source.ReSyncInterval))` (k tomu sa este vratim trocha v dalsom PR).

Tj. vysledne budem v DD vidiet napr. `60s` span orchestratora, v ktorom je dany watch (ktory spusta tasky).
Po danom intervale, sa to zase cele preskenuje, a zalozia sa tasky pre tie `keys` z prefixu, ktore este nemaju beziaci task. To zabezpecuje, ze ak nejaky task zlyha, tak sa spusti opat.